src/core.ts

```diff
@@ -137,8 +156,13 @@ export async function crawl(config: Config) {
   }
 }
 
-export async function write(config: Config) {
+export async function write(config: Config, paginationCounter: number = 0) {
   configSchema.parse(config);
+  let fileNameParts = config.outputFileName.split('.');
+  if (paginationCounter) {
+    fileNameParts.splice(fileNameParts.length - 1, 0, `${paginationCounter}`);
+  }
+  const outputFilePath = fileNameParts.join('.');
```

Comment:
@maxime4000 you are right. It used to be that the size of the document would make a difference. But with GPT they have obviously handled this better. Thanks for reviewing the PR, but I will withdraw it as I don't think it will add value now.